src/Raven.Server/RavenServer.cs

```diff
@@ -829,13 +828,18 @@ private void SendErrorIfPossible(TcpConnectionOptions tcp, Exception e)
             return stream;
         }
 
-        private bool TryAuthorize(RavenConfiguration configuration, Stream stream, TcpConnectionHeaderMessage header, out string msg)
+        private bool TryAuthorize(RavenConfiguration configuration, TcpConnectionOptions tcp, TcpConnectionHeaderMessage header, out string msg)
```

Comment:
Why are you doing this check on connection, shouldn't this be done once at startup to see if we can even bind to it?